pkg/core/client/git.go

```diff
@@ -0,0 +1,62 @@
+/**
+ * This package is intended for git access from FuseML client
+ * It needs GITEA_URL set as environment variable.
+ */
+package git
+
+import (
+	"fmt"
+	"log"
+	"net/url"
+	"os"
+	"os/exec"
+	"path"
+	"time"
+
+	"github.com/pkg/errors"
+
+	config "github.com/fuseml/fuseml-core/pkg/core/config"
+)
+
+// Push the code from local dir to remote repo
+func Push(org, name, location string) error {
+	log.Printf("Pushing the code to the git repository...")
+
+	gitUrl, exists := os.LookupEnv("GITEA_URL")
+	if !exists {
+		return errors.New("Value for gitea URL (GITEA_URL) was not provided.")
+	}
+
+	u, err := url.Parse(gitUrl)
+	if err != nil {
+		return errors.Wrap(err, "Failed to parse gitea url")
+	}
+	// TODO username+password are created by server action before git push
+	// The values could be returned from some POST if we had some init command, but
+	// for now let's assume the values are fixed
+	username := config.DefaultUserName(org)
+	password := config.DefaultUserPassword
+
+	u.User = url.UserPassword(username, password)
+	u.Path = path.Join(u.Path, org, name)
+
+	// TODO: use some real API instead...
+	cmd := exec.Command("/bin/sh", "-c", fmt.Sprintf(`
```

Comment:
Just leaving this as a note here: there are two golang git libraries that could be used to achieve this, https://github.com/go-git/go-git and https://github.com/libgit2/git2go.

Reply:
Yep, I've already noticed. But I will probably leave it for next PR.